azurerm/internal/services/containers/container_registry_resource.go

```diff
@@ -170,6 +170,23 @@ func resourceContainerRegistry() *schema.Resource {
 				},
 			},
 
+			"quarantine_policy": {
+				Type:       schema.TypeList,
+				MaxItems:   1,
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
```

Comment:
this doesn't make sense as a block - we can reduce this down to a boolean flag `quarantine_enabled` and achieve the same behaviour (erroring if this is enabled for a non-premium sku) - can we update this?

Reply:
I think it's better to keep consistent schema with API. For example the `trust_policy` below, it will add a property called `type` and this way is easier to add a new property without breaking changes.


refs:
https://docs.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries?tabs=json#TrustPolicy

Reply:
There's no need for this to be a nested block, so this can be condensed - API compatibility is helpful in some cases but in this instance doesn't make sense

Reply:
Okay, I'll update it. Thanks!